pwnlib/encoders/encoder.py

```diff
@@ -104,7 +79,7 @@ def encode(raw_bytes, avoid=None, expr=None, force=0, pcreg=''):
     msg = msg.replace('%', '%%')
     log.error(msg)
 
-all_chars        = list(chr(i) for i in range(256))
+all_chars        = list(bytes([i]) for i in range(256))
 re_alphanumeric  = r'[^A-Za-z0-9]'
```

Comment:
These patterns have to be bytes as well, to make the below functions work.
`TypeError: cannot use a string pattern on a bytes-like object`